drv/lpc55-sprot-server/src/handler.rs

```diff
@@ -568,8 +568,48 @@ impl<'a> Handler {
                 )?;
                 Ok((RspBody::Ok, None))
             }
+            ReqBody::State(StateReq::DevOrRelease) => {
+                let out = self.state_dev_or_release();
+
+                Ok((RspBody::State(out.map(StateRsp::DevOrRelease)), None))
+            }
         }
     }
+
+    fn state_dev_or_release(
+        &mut self,
+    ) -> Result<StateDevOrRelease, StateError> {
+        const CMPA_SIZE: usize = 512;
+        let mut buf = [0u8; CMPA_SIZE];
+
+        // If the SHA-256 Digest is zeros, the CMPA is unlocked.  The SHA-256
+        // digest is located in the last 4 words of the CMPA.
+        self.update
+            .read_rot_page(RotPage::Cmpa, &mut buf)
+            .map_err(StateError::ReadCmpa)?;
+        let unlocked = buf[CMPA_SIZE - 32..].iter().all(|b| *b == 0);
+
+        // Otherwise, check which root keys are enabled
+        let dev = if unlocked {
```

Comment:
The enum of RoT lifecycle states in RFD 286 identifies this configuration as `Unprogrammed`. AFAIK this work is motivated by a need to implement some policy in the SP and that policy seems to have leaked into this interface?
Regardless, I'd prefer that this interface expose the platform lifecycle states per the RFD and we can have the policy layer treat the `Unprogrammed` and `Development` states as equivalent or whatever makes sense for the policy you're trying to enforce.

Reply:
Switched to `LifecycleState` in 370c89cc